packages/devtools_app/lib/src/shared/side_panel.dart

```diff
@@ -50,23 +71,20 @@ class _ReleaseNotesViewerState extends State<ReleaseNotesViewer>
 
   late bool isVisible;
 
-  late ReleaseNotesController releaseNotesController;
-
   @override
   void didChangeDependencies() {
```

Comment:
since we are now accessing things from `widget` in `didChangeDependencies` we also need to reset the fields and listeners in `didUpdateWidget` when `widget.controller != oldWidget.controller`. Additionally, since we are no longer using context in this initialization code, we can change from `didChangeDependencies` to `initState`.

I'd create a helper method that you call from both `initState` and `didUpdateWidget`. Looks like the helper should have everything in this block besides the initialization of the visibility controller and animation